RecipeMatch/View Controllers/ProfileViewController.m

```diff
@@ -9,18 +9,96 @@
 #import "Parse/Parse.h"
 #import "LoginViewController.h"
 #import "SceneDelegate.h"
+#import "GridRecipeCell.h"
+#import "LikedRecipe.h"
+#import "UIImageView+AFNetworking.h"
 
-@interface ProfileViewController ()
-
+@interface ProfileViewController () <UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout>
+@property (weak, nonatomic) IBOutlet UICollectionView *recipesCollectionView;
+@property (nonatomic, strong) NSArray *recipes;
+@property (weak, nonatomic) IBOutlet UICollectionViewFlowLayout *flowLayout;
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
 @end
 
 @implementation ProfileViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    
+    self.recipesCollectionView.dataSource = self;
+    self.recipesCollectionView.delegate = self;
+
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self action:@selector(fetchRecipes) forControlEvents:UIControlEventValueChanged];
+    
+    if (@available(iOS 10.0, *)) {
+        self.recipesCollectionView.refreshControl = self.refreshControl;
+    } else {
+        [self.recipesCollectionView addSubview:self.refreshControl];
+    }
+    
+
+    [self fetchRecipes];
     // Do any additional setup after loading the view.
 }
 
+- (void) fetchRecipes{
+    PFQuery *recipeQuery = [LikedRecipe query];
+    [recipeQuery orderByDescending:@"createdAt"];
+    [recipeQuery includeKey:@"user"];
+    //postQuery.limit = 20;
+    [recipeQuery whereKey:@"user" equalTo:[PFUser currentUser]];
+    
+    // fetch data asynchronously
+    [recipeQuery findObjectsInBackgroundWithBlock:^(NSArray<LikedRecipe *> * _Nullable recipesFound, NSError * _Nullable error) {
```

Comment:
i'd move all of this to a separate api class, and will only call the api request from here